web/src/reducers/activities.js

```diff
@@ -53,6 +72,20 @@ const reducer = (state = initialState, action) => {
         allIds: [...state.allIds, id]
       };
     }
+    case ADD_ACTIVITY_CONTRACTOR_RESOURCE:
+      return u(
+        {
+          byId: {
+            [action.id]: {
+              contractorResources: contractors => [
+                ...contractors,
+                newContractorResource(contractors.length)
```

Comment:
I'll fix this after #448 is merged, so I can make sure they both work the same way.

Reply:
ready for you